Lib/fontTools/feaLib/error.py

```diff
@@ -6,6 +6,8 @@ def __init__(self, message, location):
         self.location = location
 
     def __str__(self):
+        if self.__cause__:
+            return self.__cause__.__str__()
```

Comment:
Hm I don't like that, this will lose/ignore the location. We need to maintain that when we re-raise from OpenTypeLibError. I suggest you do `raise FeatureLibError(str(e), location) from e` when the message that is specific to a FeatureLibError is unimportant and you want to reuse the text from the underlying OpenTypeLibError that's being handled.

Reply:
I just noticed that this OpenTypeLibError is basically identical to the original FeatureLibError and also has a location attribute. Similarly the lookup builders that were moved to otlLib also still keep a reference to their location in the feature file. I would argue that the whole concept of `location` (line and column inside a text file) is alien to otlLib.builder, and only belongs to feaLib proper. I would prefer if otlLib lookup builders were agnostic of location, and the responsibility of providing contextual info was done by feaLib, when catching OpenTypeLibError.

Reply:
I understand this, but I think it's going to be very difficult in practice. The whole point of having a "builder" class is that you parse rules and add them to a builder object, and then you build all the lookups at the end. If error only arises when the lookup is built, and the builder object isn't keeping track of the location that gave rise to the error, *what is*?

I would say that the concept of location *is* important to a builder, but `location` doesn't just mean a line-and-column-in-a-text-file. We still need to know where a lookup originally "came from" even if the user of otlLib is a graphical font editor. So what I tried to do with my previous PR was to turn `location` into a generic userdata attribute and leave its interpretation up to the user class.

Reply:
> turn location into a generic userdata attribute and leave its interpretation up to the user class.

I like this idea actually.
If the otlLib classes do not manipulate this passed-in `location` but simply use as prefix for the exception error messages, the caller that uses otlLib builders can pass in whatever they want as location, it only needs to be something that pretty-print itself nicely when `str(location)`.
So instead of passing in a tuple as location parameter from feaLib to otlLib, feaLib should pre-format the tuple as a colon-separated str, and then otlLib just interpolate that into the error messages. Let's do that.

Reply:
we do we need to do `if self.__cause__: return self.__cause__.__str__()`? I don't like that we mess with python built-ibn exception chaning mechanism. If we want FeatureLibError to print the same message as the previous exception that it is handling, then we do that e.g. `raise FeatureLibError(str(e), location) from e`